src/utils/bolus.js

```diff
@@ -129,6 +106,66 @@ export function getRecommended(insulinEvent) {
 }
 
 /**
+ * getDelivered
+ * @param {Object} insulinEvent - a Tidepool bolus or wizard object
+ *
+ * @return {Number} units of insulin delivered in this insulinEvent
+ */
+export function getDelivered(insulinEvent) {
+  let bolus = insulinEvent;
+  if (_.get(insulinEvent, 'type') === 'wizard') {
+    bolus = getBolusFromInsulinEvent(insulinEvent);
+    if (!bolus.normal && !bolus.extended) {
+      return NaN;
+    }
+  }
+  if (bolus.extended != null) {
+    if (bolus.normal != null) {
+      return fixFloatingPoint(bolus.extended + bolus.normal);
+    }
+    return bolus.extended;
+  }
+  return bolus.normal;
+}
+
+/**
+ * getDuration
+ * @param {Object} insulinEvent - a Tidepool bolus or wizard object
+ *
+ * @return {Number} duration value in milliseconds
+ */
+export function getDuration(insulinEvent) {
+  let bolus = insulinEvent;
+  if (_.get(insulinEvent, 'type') === 'wizard') {
+    bolus = getBolusFromInsulinEvent(insulinEvent);
+  }
+  // don't want truthiness here because want to return duration
+  // from a bolus interrupted immediately (duration = 0)
+  if (bolus.duration == null) {
```

Comment:
The check here is actually mostly just against `undefined`...but I used `== null` because that handles both `undefined` and `null` through coercion.

I definitely see your point about explicitness. After looking the Lodash 3.x docs again, perhaps `if (!_.inRange(bolus.duration, Infinity))` would be most explicit? That requires both finiteness and `>= 0`, which is pretty much exactly the requirements of the value (for the purposes of rendering).